### PR TITLE
more typing in the tests

### PR DESCRIPTION
**Pull request by reaperhulk**
This adds typing to `_load_cert` and now that it returns a real type we can improve our tests (and switch to using our better APIs that do more effective type narrowing in many cases).

One nuance here is that now that we allow x25519/x448 as public key types we have an odd asymmetry where it's sane to encode that in SKI but nonsensical to have it in AKI. I've left the mypy types such that this is the case (although you can defeat the typing by generating an SKI and then create the AKI from SKI. Sorry, Python typing isn't expressive enough to prevent that).

### CODE REVIEW

File: tests/x509/test_x509.py [hunk 1]
```diff
@@ -62,7 +62,10 @@ def __init__(self, value):
     value = utils.read_only_property("_value")
 
 
-def _load_cert(filename, loader, backend=None):
+T = typing.TypeVar("T")
+
+
+def _load_cert(filename, loader: typing.Callable[..., T], backend=None) -> T:
```
Comment (alex) [inline tests/x509/test_x509.py:68]: Same deal, this shouldn't be `...`, this should just be `bytes`?

File: tests/x509/test_x509.py [hunk 2]
```diff
@@ -263,6 +266,7 @@ def test_get_revoked_certificate_by_serial_number(self, backend):
         )
         serial_number = 725064303890588110203033396814564464046290047507
         revoked = crl.get_revoked_certificate_by_serial_number(serial_number)
+        assert isinstance(revoked, x509.RevokedCertificate)
```
Comment (alex) [inline tests/x509/test_x509.py:269]: Why is this required?

**Review by alex — APPROVED** [review]

File: tests/x509/test_x509.py (hunk 1 above)
Comment (reaperhulk) [reply tests/x509/test_x509.py:68]: Yes (although it'd be `[bytes]` I suppose). I skipped it for the same reason I skipped the last one: I cared exclusively about the return type and figured we could come back around and completely type it later.

File: tests/x509/test_x509.py (hunk 2 above)
Comment (reaperhulk) [reply tests/x509/test_x509.py:269]: `get_revoked_certificate_by_serial_number ` returns `Optional[RevokedCertificate]` so we need to assert on it. We can do isinstance or we could just assert not None, but otherwise the next line will fail
